src/DateTimePicker.jsx

```diff
@@ -0,0 +1,240 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import cx from 'classnames';
+import CalendarComponent from './CalendarComponent';
+import TimeInput from './TimeInput';
+import DateTimeLocalInput from './DateTimeLocalInput';
+
+/**
+ * @module DateTimePicker
+ * @description a component that renders a calendar ui and time input
+ * defaults to datetime-local input on supported mobile browsers.
+ */
+class DateTimePicker extends React.Component {
+
+	constructor(props) {
+		super(props);
+		this.state = {
+			datetime: props.value ? new Date(props.value) : '',
+			isDateTimeLocalSupported: false
+		};
+
+		this.getDate = this.getDate.bind(this);
+		this.setDate = this.setDate.bind(this);
+
+		this.getTime = this.getTime.bind(this);
+		this.setTime = this.setTime.bind(this);
+
+		this.setDateTime = this.setDateTime.bind(this);
+		this.onFocus = this.onFocus.bind(this);
+		this.onBlur = this.onBlur.bind(this);
+	}
+
+	componentWillMount() {
+		this.setState({ isDateTimeLocalSupported: this.hasBrowserSupport() });
+	}
+
+	/**
+	* @function hasBrowserSupport
+	* @description test if this browser supports datetime local
+	* @return bool whether or not this browser supports datetime local
+	*/
+	hasBrowserSupport() {
+		const input = document.createElement('input'),
+			invalidValue = 'notadate';
+
+		// some browsers (as Android stock browsers) pretend they support
+		// certain input types,
+		// so we set the value to an invalid value and see if browser rejects
+
+		input.setAttribute('type', 'datetime-local');
+		input.setAttribute('value', invalidValue);
+
+		return !(this.props.forceCalendar || input.value === invalidValue);
+	}
+
+	/**
+	* @function getDate
+	* @description gets a date from the datetime value in state
+	* @return Date
+	*/
+	getDate() {
+		if (!this.state.datetime) {
+			return;
+		}
+		const datetime = new Date(this.state.datetime);
+		return new Date(datetime.getFullYear(), datetime.getMonth(), datetime.getDate());
+	}
+
+	/**
+	* @function setDate
+	* @description sets the date portion of the state's datetime
+	* @param Date value date to set in state
+	*/
+	setDate(value) {
+		const newDate = new Date(value);
+		const datetime = new Date(this.state.datetime);
+
+		datetime.setFullYear(newDate.getFullYear(), newDate.getFullMonth(), newDate.getDate());
+		this.setState({ datetime });
+	}
+
+	/**
+	* @function parseTimeFromDateTime
+	* @description gets the time string part of our datetime state
+	* @return String
+	*/
+	parseTimeFromDateTime(datetime) {
+		return (datetime.toTimeString().split(' ')[0]).split(':', 2).join(':'); // TODO localize toLocaleTimeString ?
```

Comment:
this part always hangs me up about this piece, it's probably because of the pickers we are using, or maybe how datetime-locale gets data. But would be good to add some doc's or @see references to why we need to do this. Either here or at the top of the component for future dev's